packages/turf-simplify/index.ts

```diff
@@ -147,11 +146,20 @@ function simplifyPolygon(
     }
     let ringTolerance = tolerance;
     let simpleRing = simplifyJS(ring, ringTolerance, highQuality);
-    // remove 1 percent of tolerance until enough points to make a triangle
-    while (!checkValidity(simpleRing)) {
+
+    // If simplified ring isn't valid (has been over simplified) reduce the
+    // tolerance by 1% and try again.
+    while (!checkValidity(simpleRing) && ringTolerance >= Number.EPSILON) {
```

Comment:
I was worried about this leaving impossibly tiny polygons with a large number of points, but EPSILON is so much smaller than any meaningful coordinate delta that this is probably a nonissue.

```
> area({type: "Polygon", coordinates: [[0,0], [0, Number.EPSILON], [Number.EPSILON, Number.EPSILON], [Number.EPSILON, 0], [0, 0]]})
0
```